test/provider/updateTabProperties.test.ts

```
@@ -0,0 +1,95 @@
+import {test} from 'ava';
+import {Fin} from 'hadouken-js-adapter';
+import {_Window} from 'hadouken-js-adapter/out/types/src/api/window/window';
+
+import {WindowIdentity} from '../../src/provider/model/DesktopWindow';
+import {executeJavascriptOnService} from '../demo/utils/serviceUtils';
+import {updateTabProperties} from '../demo/utils/tabServiceUtils';
+
+import {getConnection} from './utils/connect';
+import {tabWindowsTogether} from './utils/tabWindowsTogether';
+import {WindowInitializer} from './utils/WindowInitializer';
+
+let fin: Fin;
+
+let wins: _Window[] = [];
+
+const windowInitializer = new WindowInitializer();
+
+test.before(async () => {
+    fin = await getConnection();
+});
+test.beforeEach(async () => {
+    wins = await windowInitializer.initWindows(2);
+});
+
+test.afterEach.always(async () => {
+    // Try and close all the windows.  If the window is already closed then it will throw an error which we catch and ignore.
+    await Promise.all(wins.map(win => {
+        try {
+            return win.close();
+        } catch (e) {
```

This will only catch any exceptions that happen synchronously within the `close` function.

To suppress any errors due to the window already being closed, will need to add a `.catch` to the `close` promise. Could also catch at the top-level after the `all`, but that could in theory start the next test before all windows have had chance to close.

```
await Promise.all(wins.map(win => {
    return win.close().catch(() => {});
}));
```